packages/canary-web-components/src/components/ic-date-input/ic-date-input.tsx

```diff
@@ -0,0 +1,1547 @@
+      input.addEventListener("input", this.handleInput);
+      input.addEventListener("focus", this.handleFocus);
+      input.addEventListener("blur", this.handleBlur);
+    });
+
+    // Prevent asterisk being read out after the label by screen reader (by applying aria-hidden)
+    // Needed because label is included in 'aria-labelledby' instead of using 'aria-label'
+    const labelEl = this.el.shadowRoot.querySelector("label");
+    if (this.required) {
+      const asteriskSpan = document.createElement("span");
+      asteriskSpan.setAttribute("aria-hidden", "true");
+      asteriskSpan.textContent = " *";
+      labelEl?.appendChild(asteriskSpan);
+    }
+  }
+
+  componentWillUpdate(): void {
+    if (!this.isDateSetFromKeyboardEvent) {
+      this.setDate(this.value);
+    }
+    this.setAriaInvalid(
+      this.isValidDay,
+      this.isValidMonth,
+      this.isValidDate,
+      this.isDisabledDate
+    );
+    this.handleDateChange(false);
+    this.isDateSetFromKeyboardEvent = false;
+  }
+
+  componentDidRender(): void {
+    this.setAriaLabelledBy();
+  }
+
+  /**
+   * Returns the value as a Date object
+   * @returns Date
+   */
+  @Method()
+  async getDate(): Promise<Date> {
+    return this.selectedDate;
+  }
+
+  /**
+   * @internal Sets focus on the calendar button.
+   */
+  @Method()
+  async setCalendarFocus(): Promise<void> {
+    if (this.calendarButtonEl) {
+      this.calendarButtonEl.focus();
+    }
+  }
+
+  /**
+   * @internal Used to pass disabledDays from parent component.
+   */
+  @Method()
+  async setDisableDays(days: IcWeekDays[]): Promise<void> {
+    this.disableDays = days;
+  }
+
+  private setInputPasteValue = (input: EventTarget, pastedValue: string) => {
+    const inputEl = input as HTMLInputElement;
+
+    inputEl.value = pastedValue;
+    inputEl.classList.add(this.FIT_TO_VALUE);
+  };
+
+  private checkSingleCopiedValueIsValid = (
+    input: EventTarget,
+    pastedValue: string
+  ) => {
+    let isValid = false;
+    const value = Number(pastedValue);
+
+    switch (input) {
+      case this.dayInputEl:
+        isValid = value >= 1 && value <= this.MAX_DAY;
+        break;
+      case this.monthInputEl:
+        isValid = value >= 1 && value <= this.MAX_MONTH;
+        break;
+      case this.yearInputEl:
+        isValid = true;
+        break;
+      default:
+        isValid = false;
+        break;
+    }
+
+    return isValid;
+  };
+
+  /**
+   * Using Object.prototype.string to determine type if event from user
+   * @param event - event object used to differentiate keyboard, generic or input event
+   * @returns boolean
+   */
+  private isKeyboardOrEvent = (event: Event) => {
+    return (
+      Object.prototype.toString.call(event) === this.EVENT_OBJECT_STRING ||
+      Object.prototype.toString.call(event) ===
+        this.KEYBOARD_EVENT_OBJECT_STRING
+    );
+  };
+
+  private setDateValidity = () => {
+    if (!isEmptyString(this.day)) {
+      if (+this.day > this.MAX_DAY) {
```

Comment:
Couple more opportunities to replace with ternary operators here. this.isValidDay and this.isValidMonth